ocfweb/docs/docs/services/hpc.md

```diff
@@ -0,0 +1,115 @@
+
+* `srun`: Used to submit jobs.
+* `scontrol`: Used to view and edit the Slurm configuration.
+* `squeue`: Used to view running and queued jobs.
+
+## Dependencies
+For managing application dependencies, you currently have two options:
+
+### Virtual Environments
+
+First you could use a [virtual environment][venv] if you are using Python packages. To create a virtual environment navigate to your home directory and run the following commands:
+
+```
+virtualenv --system-site-packages -p python3 venv
+. venv/bin/activate
+```
+
+This will allow you to *pip install* any Python packages that the OCF does not already have for your program.
+
+### Singularity
+
+For those who need access to non-Python dependencies or have already integrated their program into Docker, the second option is to use Singularity containers. [Singularity][singularity] is a containerization platform developed at Lawrence Berkeley National Laboratory that is designed specifically for HPC environments. To read more about the benefits of Singularity you can look [here][singularity_article]. We suggest a particular workflow, which will help simplify deploying your program on our infrastructure.
+
+#### Installing
+
+If you are running linux on your system, you can install Singularity from the official apt repos:
+
+```
+sudo apt install singularity-container
+```
+
+If you are running Mac you can look [here][mac_install], or Windows [here][win_install].
+
+#### Building Your Container
+
+```
+singularity build --sandbox ./my_container docker://ubuntu
+```
+This will create a Singularity container named `my_container`. If you are still testing out your container on your system, we suggest you use the `--sandbox` option. This will allow you to install new packages in your container as you need them without needing sudo privileges. The `docker://ubuntu` option notifies Singularity to bootstrap the container from the official Ubuntu docker container on [Docker Hub][docker_hub]. There is also a [Singularity Hub][singularity_hub], from which you can directly pull Singularity images in a similar fashon. We also have some pre-built containers that you may use to avoid having to build your own. They are currently located at `/home/containers`.
+
+#### Using Your Container
+
+```
+singularity shell my_container
+```
+The above command will allow you to shell into your container. By default your home directory in the container is linked to your real home directory outside of the container environment, which helps you avoid having to transfer files in and out of the container.
+
+```
+singularity exec --nv my_container ./my_executable.sh
+```
+This command will open your container and run the `my_executable.sh` script in the container environment. The `--nv` option allows the container to interface with the GPU. This command is useful when using `srun` so you can run your program in a single command.
```

Comment:
what is meant by "allows the container to interface with the GPU"? How does it allow this?

Reply:
lol if i knew i would tell you, the option is not included in the documentation or on the man page. The only reference to it is [here](http://singularity.lbl.gov/docs-exec#a-gpu-example)

Reply:
Likely it bind-mounts /dev/nvidiactl into the container. In any case, it's not relevant to this doc.